layers/cip.go

```diff
@@ -0,0 +1,138 @@
+// Copyright 2017 Google, Inc. All rights reserved.
+//
+// Use of this source code is governed by a BSD-style license
+// that can be found in the LICENSE file in the root of the source
+// tree.
+
+package layers
+
+import (
+	"encoding/binary"
+	"errors"
+
+	"github.com/google/gopacket"
+)
+
+const (
+	cipBasePacketLen int = 2
+)
+
+var (
+	ErrCIPDataTooSmall = errors.New("CIP packet data truncated")
+)
+
+type CIP struct {
```

Comment:
Add a comment describing this, potentially linking to RFCs or the like.

Reply:
Unfortunately there aren't any RFCs (2653 is a different CIP) and the meaning of most fields is dependent on the manufacturer; I used https://www.rockwellautomation.com/resources/downloads/rockwellautomation/pdf/sales-partners/technology-licensing/eipexp1_2.pdf and similar documents to get the formatting of the messages, but the meaning of each code is going to be different on a Rockwell device vs. a Datalogic device. I can put in a comment linking to that PDF, which at least contains a description of the basic protocol

Reply:
That's hilarious, I really thought "Rockwell Automation" was just a made-up name for this: https://www.youtube.com/watch?v=vHk40VvFsp4

Given the lack of RFC, instead let's just describe what this actually is ("CIP implements encoding/decoding for the Common Industrial Protocol, used by XXX in industry YYY for ZZZ...")